extensions/ql-vscode/test/vscode-tests/minimal-workspace/variant-analysis/ghec-dr.test.ts

```diff
@@ -0,0 +1,44 @@
+import { ConfigurationTarget } from "vscode";
+import {
+  VARIANT_ANALYSIS_ENABLE_GHEC_DR,
+  VSCODE_GITHUB_ENTERPRISE_URI_SETTING,
+} from "../../../../src/config";
+import { isVariantAnalysisEnabledForGitHubHost } from "../../../../src/variant-analysis/ghec-dr";
+
+describe("checkVariantAnalysisEnabled", () => {
+  it("returns cleanly when no enterprise URI is set", async () => {
+    expect(isVariantAnalysisEnabledForGitHubHost()).toBe(true);
+  });
+
+  it("returns false when GHES enterprise URI is set", async () => {
+    await VSCODE_GITHUB_ENTERPRISE_URI_SETTING.updateValue(
+      "https://github.example.com",
+      ConfigurationTarget.Global,
+    );
+    await VARIANT_ANALYSIS_ENABLE_GHEC_DR.updateValue(
+      "true",
+      ConfigurationTarget.Global,
+    );
+    expect(isVariantAnalysisEnabledForGitHubHost()).toBe(false);
+  });
+
+  it("returns false when GHEC-DR URI is set but variant analysis feature flag is not set", async () => {
+    await VSCODE_GITHUB_ENTERPRISE_URI_SETTING.updateValue(
+      "https://github.example.com",
+      ConfigurationTarget.Global,
+    );
+    expect(isVariantAnalysisEnabledForGitHubHost()).toBe(false);
+  });
+
+  it("returns true when GHEC-DR URI is set and variant analysis feature flag is set", async () => {
```

Comment:
Should we also have a test for a GHES-DC URI but without the feature flag set?

Reply:
I think that's covered by the "returns false when GHEC-DR URI is set but variant analysis feature flag is not set" test just above this one. Am I understanding correctly?

Reply:
Ah, I should have left a comment on that test. I was looking at the contents of the test rather than the names, but the test above this one uses `https://github.example.com` rather than a GHEC-DR URI.

Reply:
Ah good spot, the URI was incorrect in that test. I'll fix that. I'll also add another test to cover the case of a GHES URI and feature flag not being set, so ever case is covered.